File: backend-dotnet/Application/Common/PostIndex.cs
using Domain.Enums;

namespace Application.Common
{
    public class PostIndex
    {
        public Guid Id { get; set; }
        public Guid UserId { get; set; }
        public string Caption { get; set; } = string.Empty;
        public List<string> Tags { get; set; } = new();
        public string Username { get; set; } = string.Empty;
        public string? ImageUrl { get; set; }
        public DateTime CreatedAt { get; set; }
        public PostVisibility Visibility { get; set; }
    }
}

File: backend-java/src/main/java/com/jerrygram/application/common/PostIndex.java
package com.jerrygram.application.common;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.elasticsearch.annotations.Document;
import org.springframework.data.elasticsearch.annotations.Field;
import org.springframework.data.elasticsearch.annotations.FieldType;

import java.time.LocalDateTime;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Document(indexName = "posts")
public class PostIndex {
    
    @Id
    private String id;
    
    @Field(type = FieldType.Text, analyzer = "standard")
    private String caption;
    
    @Field(type = FieldType.Text)
    private String imageUrl;
    
    @Field(type = FieldType.Keyword)
    private String authorId;
    
    @Field(type = FieldType.Text)
    private String authorUsername;
    
    @Field(type = FieldType.Keyword)
    private String visibility;
    
    @Field(type = FieldType.Integer)
    private Integer likesCount;
    
    @Field(type = FieldType.Integer)
    private Integer commentsCount;
    
    @Field(type = FieldType.Keyword)
    private List<String> tags;
    
    @Field(type = FieldType.Date)
    private LocalDateTime createdAt;
    
    @Field(type = FieldType.Boolean)
    private Boolean isActive;
}